docs/apis/calcite.md

```diff
@@ -0,0 +1,570 @@
+Apache Apex, unified stream processing engine enables application developers to process data at very high throughput with low latency. Though type of data demands different type of
+processing requirements but SQL remains way to process input data .To ensure that existing ETL developers and developers who are well versed with Database applications adopt stream processing application development with ease, integration of SQL with Apex was needed. Being popular Apache project, Apache Calcite was chosen for an integration and to facilitate use of SQL with Apex.
+
+## APEX-Calcite Integration
+
+ Apache Calcite is highly customizable engine for parsing and planning queries on relational data from various data sources . Apache Calcite provides functionality of storage independent query optimization and ways to integrate them in other frameworks which would like to take advantage and expose SQL capability to their users.
+
+Particularly in Apex-Calcite integration, Calcite processes query and then creates Relational Algebra to create processing pipelines. These relational algebra processing pipelines are converted to DAG with set of operators to perform business logic on stream data.
+
+![image alt text](images/calcite-apex.png)
+
+
+Above figure explains how SQL query gets converted to Apex DAG.
+
+1. The received query from a user is parsed by Calcite’s query parser.
+2. The query parser returns the SQL Node Tree to Query Planner and optimizer which converts the SQL Node Tree to Relation Expression Tree.
```

Comment:
"Query Planner & Optimizer"... Put in quotes or make it italic. As of now "and" in it looks like end of sub-sentense.